Messaging #87

eric-silva-61 opened the pull request:
### What this PR does (required):
- Added MessagingContext
- Context transforms API data to match what UI expects
- Adjusted BE API to return all conversation and messages at once
- Updates to UI to use new context

### Screenshots / Videos (required):
![image](https://user-images.githubusercontent.com/62892917/128072226-75820c91-b217-4645-80f5-8ce12207043b.png)

### Any information needed to test this feature (required):
- Go to messaging tab using the demo login

### Any issues with the current functionality (optional):
-none

eric-silva-61 committed:
Updated server controller/models/routes.  Added msg context

eric-silva-61 committed:
Conversation models adjusted and context updated to transform API data to UI shape

rajivtitus commented on client/src/App.tsx:
```diff
@@ -30,25 +31,27 @@ function App(): JSX.Element {
+                    <ProtectedRoute exact path="/setting" component={ProfileSetting} />
+                    {/* Replace component with Discovery Page */}
+                    <Route exact path="/discovery" component={Discovery} />
+                    {/* Replace the components once created */}
```
As we won't be adding any more new components, please feel free to remove the comments here. Thanks!

eric-silva-61 replied:
Done

rajivtitus commented on client/src/context/useMessagingContext.tsx:
```diff
@@ -0,0 +1,57 @@
+
+export const MessagingProvider: FC = ({ children }): JSX.Element => {
+  const { loggedInUser } = useContext(AuthContext);
+  const [conversations, setConversations] = useState<Conversation[]>([]);
```
How are we planning on updating and displaying the messages within a conversation? Also, how are we going to differentiate which conversation is the active one?

eric-silva-61 replied:
Added add message functionality to context, assuming that with the socket io integration the backend would update the database.  Added indicator for active conversation.
![image](https://user-images.githubusercontent.com/62892917/128221800-e48d3de5-cb7f-4b96-8126-c11f13b18af2.png)

eric-silva-61 committed:
PR comments, MUI deprecation fix, key fix

eric-silva-61 committed:
Styling added to indicated active conversation

eric-silva-61 committed:
Handle empty conversations

eric-silva-61 committed:
Added create new conversation functionality

eric-silva-61 commented:
Also added in new conversation functionality.  

- If conversation already exists returns id
- if new conversation, creates it on server and returns new id
- added route (/messages/:id) to handle going to messages and loading correct conversation

eric-silva-61 committed:
Added routing to load message (new or current)

eric-silva-61 committed:
Merge branch 'main' into messaging-#87

eric-silva-61 committed:
Cleaned up unused useEffect

eric-silva-61 committed:
Merge branch 'main' into messaging-#87

rajivtitus commented on client/src/context/useMessagingContext.tsx:
```diff
@@ -0,0 +1,128 @@
+  newConversation: () => null,
+});
+
+// const newConversation = (conversations: APIConversation[], loggedInUserEmail: string): Conversation[] =>
```
Please remove commented code if we are no longer using it

rajivtitus reviewed (approved):
LGTM! Thanks for making requested changes

eric-silva-61 committed:
Merge branch 'main' into messaging-#87
